onnxruntime/core/optimizer/layer_norm_fusion.cc

```diff
@@ -117,6 +130,47 @@ Status LayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
     nodes_to_remove.push_back(sub_node);
 
+    // Check if the ReduceMean and Sub nodes have same input, and if that input is a Cast
+    // node, we can also remove it.
+    const NodeArg* p_reduce_mean_input = reduce_mean_node.MutableInputDefs()[0];
+    const NodeArg* p_sub_input = nullptr;
+    for (NodeArg* node_arg : sub_node.MutableInputDefs()) {
+      if (node_arg != reduce_mean_node.MutableOutputDefs()[0]) {
+        p_sub_input = node_arg;
+        break;
+      }
+    }
+
+    if (!p_reduce_mean_input || !p_sub_input || p_reduce_mean_input != p_sub_input) {
+      continue;
+    }
+
+    if (p_sub_node_dup) {
+      const NodeArg* p_sub_dup_input = nullptr;
+      for (NodeArg* node_arg : graph.GetNode(p_sub_node_dup->Index())->MutableInputDefs()) {
+        if (node_arg != reduce_mean_node.MutableOutputDefs()[0]) {
+          p_sub_dup_input = node_arg;
+          break;
+        }
+      }
+      if (!p_sub_dup_input || p_reduce_mean_input != p_sub_dup_input) {
+        continue;
+      }
+    }
+
+    const Node* p_reduce_mean_input_node = graph_utils::GetInputNode(reduce_mean_node, 0);
+    bool has_leading_cast = false;
+    if (p_reduce_mean_input_node) {
+      Node& reduce_mean_input_node = *graph.GetNode(p_reduce_mean_input_node->Index());
+      // If input to Pow is a Cast, and the Cast has same consumer count as subCnt + 1
+      if (graph_utils::IsSupportedOptypeVersionAndDomain(reduce_mean_input_node, "Cast", {9, 13}) &&
```

Comment:
don't need check Pow op or not?

Reply:
Wrong comment... Nothing to do with Pow, should not be Pow here, but the first ReduceMean.